components/scrollable-container/script.js

```diff
@@ -33,12 +37,27 @@ class ScrollableContainer extends HTMLElement {
                 // render the component
                 components.renderOnce(this);
                 // do the initial setup - add event listeners, assign members
+
                 this.setup();
                 this.shouldShowScrollbar();
+                if (this.automatic) this.initMutationObserver();
             })
             .catch(err => console.error(err));
     }
 
+    initMutationObserver() {
+        const observer = new MutationObserver(() => this.shouldShowScrollbar());
+
+        const options = {
+            attributes: true,
+            subtree: true,
+            childList: true,
+            attributesFilter: ['style', 'class']
+        }
+
+        observer.observe(this, options);
```

Comment:
Should we stop listening at some point?

Reply:
1. Should we add a setter for setting the automatic property also?
2. If the `automatic` attribute is removed, we need to stop the current MutationObserver and either:
2.1. Add a new MutationObserver with options to only watch for attribute changes and handle the `automatic` attribute existance and re-add the other observer.
2.2 If it works in Cohtml, we can utilize the [attributeChangedCallback](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#using_the_lifecycle_callbacks) and enable <-> disable the mutation observer then.

Reply:
makes sense. added attributeChangedCallback